Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1062 @@
+                {
+                    $frontdoorEndpoint.isMinTLSVersionRolledBackCustomDomainPostRemediation = $true
+                    $frontdoorEndpoint.isMinTlsVersionSetOnCustomDomain = $false
+                    $frontdoorEndpoint.minimumTlsVersion = $requiredMinTLSVersion
+                    $frontDoorsRolledBack += $frontdoorEndpoint
+                    Write-Host "Minimum required TLS version for Front Door has been Rolled back successfully for FrontDoor" ($resourceName) "with Domain" ($DomainName) -ForegroundColor $([Constants]::MessageType.Update)
+                    Write-Host $([Constants]::SingleDashLine)
+                }
+                else
+                {
+                    $frontDoorsEPSkipped += $frontdoorEndpoint
+                    $logResource = @{}
+                    $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                    $logResource.Add("ResourceName",($_.ResourceName))
+                    $logResource.Add("Reason", "Error while Rolling back the minimum required TLS version. Skipping this Front Door.")
+                    $logSkippedResources += $logResource
+                    Write-Host "Error while Rolling back the minimum required TLS version for this Front Door." -ForegroundColor $([Constants]::MessageType.Error)
+                    Write-Host "Skipping this Front Door." -ForegroundColor $([Constants]::MessageType.Warning)
+                    Write-Host $([Constants]::SingleDashLine)
+                    return
+                }
+                    
+            }
+            catch
+            {
+                $frontDoorsEPSkipped += $frontdoorEndpoint
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason", "Error while resetting the minimum required TLS version on the custom domain. Skipping this Front Door.")
+                $logSkippedResources += $logResource
+                Write-Host "Error while resetting the minimum required TLS version on the custom domain of Front Door." -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::SingleDashLine)
+                return
+            }   
+        }
+
+        Write-Host "Successfully rolled back changes on the Front Door." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+
+    if (($frontDoorsEPSkipped | Measure-Object).Count -eq 0)
+    {
+        Write-Host "Minimum TLS Version successfully reset on the for all [$($totalFrontDoors)] Front Door Endpoint(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else
+    {
+        Write-Host "Minimum TLS Version successfully reset for [$($($frontDoorsRolledBack | Measure-Object).Count)] out of [$($totalFrontDoors)] Front Door(s)" -ForegroundColor $([Constants]::MessageType.Warning)
```

Comment:
NIT: here should we say front door or front door endpoints.

Reply:
Renamed to endpoints